tests/validator_test.go

```diff
@@ -329,4 +338,618 @@ var _ = Describe("Template validator", func() {
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:2960] Negative test - Start a VM with machine type violation", func() {
+			template = TemplateWithRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			// set value unfulfilling validation
+			vmi = addDomainResourcesToVMI(vmi, 2, "test", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Should have given error failing to fulfill validations")
+		})
+		It("test with template optional rules unfulfilled", func() {
+			template = TemplateWithRulesOptional()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("test with template rules unfulfilled", func() {
+			template = TemplateWithRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Should have given error failing to fulfill validations")
+		})
+		It("Test template with incorrect rules satisfied", func() {
+			template = TemplateWithIncorrectRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Should have given error failing to fulfill validations")
```

Comment:
The same comment as above, you need to check the error.

Reply:
Added error type check